### ISSUE TITLE
Shields of Norway

### ISSUE THREAD

**Issue by claysmalley** [ZeLonewolf/openstreetmap-americana#1110]
Norway posts two networks of routes: national roads _([riksveier](https://no.wikipedia.org/wiki/Riksvei))_ and county roads _([fylkesveier](https://no.wikipedia.org/wiki/Fylkesvei))._ National roads have a white-on-green rectangular shield, matching that of the E-road network. County roads also have a rectangular shield, but black-on-white.

Before shields can be supported in Americana, the Norwegian mapping community needs to come to a consensus on how to tag route relations.

[![image](https://upload.wikimedia.org/wikipedia/commons/b/b9/Vegskilt_Sogndal.jpg)](https://commons.wikimedia.org/wiki/File:Vegskilt_Sogndal.jpg)

**Comment by ZeLonewolf** [comment]
Is the dashed border something we need to support?

**Comment by claysmalley** [comment]
> Is the dashed border something we need to support?

No, it's the same thing as [this](https://github.com/ZeLonewolf/openstreetmap-americana/pull/438#issuecomment-1160687463). Nordic countries use a dotted-line border where US states would use a "TO" banner.

**Comment by hohMiyazawa** [comment]
Routes in Norway are mapped with ``ref`` on the highway segments. There is little motivation in the community for making these into relations.

All refs starting with an "E" (Europaveier) use green shields, the other national roads have no tagging identifying them as such except for the ref number, which would have to be looked up in a list of ~50 numbers. (They do not change often).

**Comment by claysmalley** [comment]
For Americana, we've chosen to use route relations instead of ways when determining what shields to display, with limited exceptions where it would be technically unfeasible. Currently, the exceptions are Ireland and the UK, where shield appearance is determined by the classification of the road (i.e. the value of `highway=*`), rather than the number. Mapping these distinctions in relations would involve laborious, redundant micromapping, thus it would not be a good fit in these countries. It helps us that they're situated on islands (see discussion [here](https://github.com/ZeLonewolf/openstreetmap-americana/issues/122#issuecomment-2142968806)). But generally, there needs to be a strong case that route relations would be insufficient or cumbersome for us to consider an alternate technical solution.

Americana already supports the E-road network—it recognizes relations tagged `route=road` + `network=e-road` and assigns them green rectangular shields accordingly. E-road relations seem to be completely mapped in most countries where they're present (even in Ireland and the UK), but there is a mysterious Norway-sized gap.

**Comment by hohMiyazawa** [comment]
Since the E-roads are fairly standard, I suppose ``route=road`` + ``network=e-road`` would be the way to go about it in Norway too.

Focusing on creating relations for those would give 80% of the benefit for 20% of the work of doing everything.

**Comment by ZeLonewolf** [comment]
Forum post:
https://community.openstreetmap.org/t/gjennsyn-med-e-road-relasjonane/115310/2

**Comment by hohMiyazawa** [comment]
A couple of the more active members have commented there, and road relations are not something they are positive to.

**Comment by ZeLonewolf** [comment]
As I [noted](https://community.openstreetmap.org/t/gjennsyn-med-e-road-relasjonane/115310/4) on that thread, we style highway shields based on OpenMapTiles schema, specifically the `route_N_*` [attributes](https://openmaptiles.org/schema/#transportation_name) in the `transportation_name` layer. So, the Norway community would either have to come to a consensus on mapping route relations or else someone would need to figure out how to modify OpenMapTiles in order to create a special case for Norway.

**Comment by ZeLonewolf** [comment]
Based on discussion in:
* [Delete E-road relations and int_ref](https://community.openstreetmap.org/t/delete-e-road-relations-and-int-ref/93213)
* [Gjennsyn med E-road relasjonane?](https://community.openstreetmap.org/t/gjennsyn-med-e-road-relasjonane/115310)

There is no consensus in Norway to map road relations of any kind, and in fact they have deleted them where they previously existed per the first thread.  Norway is an outlier in that they are the only country which does not have E-road route relations.  Given that, we are not able to support highway shields in Norway.
